src/main/java/com/whiteclarkegroup/liquibaselinter/config/rules/RuleRunner.java

```diff
@@ -1,47 +1,86 @@
 package com.whiteclarkegroup.liquibaselinter.config.rules;
 
 import com.whiteclarkegroup.liquibaselinter.ChangeLogParseExceptionHelper;
+import com.whiteclarkegroup.liquibaselinter.config.Config;
 import liquibase.change.Change;
 import liquibase.changelog.DatabaseChangeLog;
 import liquibase.exception.ChangeLogParseException;
+import org.springframework.context.annotation.ClassPathScanningCandidateComponentProvider;
+import org.springframework.core.type.filter.AssignableTypeFilter;
 
-import java.util.Arrays;
-import java.util.Map;
-import java.util.Optional;
+import java.util.*;
+import java.util.stream.Collectors;
 
 public class RuleRunner {
+    private final Config config;
+    private final List<ChangeRule> changeRules;
 
-    private final Map<String, RuleConfig> ruleConfigs;
+    public RuleRunner(Config config) {
+        this.config = config;
+        this.changeRules = discoverChangeRules();
+    }
 
-    public RuleRunner(Map<String, RuleConfig> ruleConfigs) {
-        this.ruleConfigs = ruleConfigs;
+    private List<ChangeRule> discoverChangeRules() {
+        final ClassPathScanningCandidateComponentProvider provider = new ClassPathScanningCandidateComponentProvider(false);
```

Comment:
Would really like to have to avoid having this code and the additional spring context dependency. I wonder if we could use similar pattern Liquibase is using its self to find our custom extension. Looks like `DefaultPackageScanClassResolver` might be a good bet.

Reply:
Good point, fixed.